RabbitMQ connector throws backpressure errors if queue is not declared during first connect attempt

If using the RabbitMQ connector with a non-declared queue, and the queue is not declared when the connector attempts to connect, the connector fails and will then continue to try and reconnect as configured.

If you then externally define the queue and wait for the connector to eventually reconnect, the connector appears ready to receive messages.

Finally, after the connector appears connected and stable, upon sending some messages, the connector stream will throw a `BackPressureFailure` with the message `Buffer full, cannot emit item`.

After the `BackPressureFailure` is thrown the connector silently stops receiving messages and continues to report itself as healthy; effectively hanging messages processing.

A log of the startup of the an example Quarkus application that uses the `StartupEvent` to declare the required exchanges and queues (as might be done in complex configurations). You can see at the end the connector(s) eventually connect properly with the late declared exchanges/queues.

<details>

```log
2022-06-23 11:49:48,096 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-60) Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2022-06-23 11:49:48,415 INFO  [io.qua.sma.dep.processor] (build-31) Configuring the channel 'reactive' to be managed by the connector 'smallrye-rabbitmq'
2022-06-23 11:49:48,422 INFO  [io.qua.sma.dep.processor] (build-31) Configuring the channel 'coroutine' to be managed by the connector 'smallrye-rabbitmq'
2022-06-23 11:49:48,423 INFO  [io.qua.sma.dep.processor] (build-31) Configuring the channel 'out' to be managed by the connector 'smallrye-rabbitmq'
2022-06-23 11:49:48,706 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-60) Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
2022-06-23 11:49:48,712 INFO  [org.tes.DockerClientFactory] (build-60) Docker host IP address is localhost
2022-06-23 11:49:48,752 INFO  [org.tes.DockerClientFactory] (build-60) Connected to docker: 
  Server Version: 20.10.14
  API Version: 1.41
  Operating System: Docker Desktop
  Total Memory: 7936 MB
2022-06-23 11:49:48,771 INFO  [org.tes.uti.ImageNameSubstitutor] (build-60) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2022-06-23 11:49:48,893 INFO  [🐳 .3.3]] (build-60) Creating container for image: testcontainers/ryuk:0.3.3
2022-06-23 11:49:48,987 INFO  [org.tes.uti.RegistryAuthLocator] (build-60) Credential helper/store (docker-credential-desktop) does not have credentials for https://index.docker.io/v1/
2022-06-23 11:49:49,107 INFO  [io.qua.arc.pro.BeanProcessor] (build-3) Found unrecommended usage of private members (use package-private instead) in application beans:
	- Bean constructor void <init>() on org.acme.ExternalService
2022-06-23 11:49:49,148 INFO  [🐳 .3.3]] (build-60) Container testcontainers/ryuk:0.3.3 is starting: e8f005b780d987b77e1eb42adb0005db7649bc2f28bfb905153ca3d20a1676a9
2022-06-23 11:49:49,454 INFO  [🐳 .3.3]] (build-60) Container testcontainers/ryuk:0.3.3 started in PT0.677642S
2022-06-23 11:49:49,464 INFO  [org.tes.uti.RyukResourceReaper] (build-60) Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
2022-06-23 11:49:49,464 INFO  [org.tes.DockerClientFactory] (build-60) Checking the system...
2022-06-23 11:49:49,466 INFO  [org.tes.DockerClientFactory] (build-60) ✔︎ Docker server version should be at least 1.6.0
2022-06-23 11:49:49,481 INFO  [🐳 .9-management]] (build-60) Creating container for image: rabbitmq:3.9-management
2022-06-23 11:49:49,543 INFO  [🐳 .9-management]] (build-60) Container rabbitmq:3.9-management is starting: 649d9ee7b1df682bbf0c7964e72b727c3e47fd424043da24240852d968c5cb8e
2022-06-23 11:49:54,104 INFO  [🐳 .9-management]] (build-60) Container rabbitmq:3.9-management started in PT4.622961S
2022-06-23 11:49:54,106 INFO  [io.qua.sma.rea.rab.dep.RabbitMQDevServicesProcessor] (build-60) Dev Services for RabbitMQ started.
2022-06-23 11:49:54,107 INFO  [io.qua.sma.rea.rab.dep.RabbitMQDevServicesProcessor] (build-60) Other Quarkus applications in dev mode will find the broker automatically. For Quarkus applications in production mode, you can connect to this by starting your application with -Drabbitmq-host=localhost -Drabbitmq-port=56268 -Drabbitmq-username=guest -Drabbitmq-password=guest
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2022-06-23 11:49:55,131 INFO  [io.sma.rea.mes.rabbitmq] (Quarkus Main Thread) SRMSG17036: RabbitMQ broker configured to localhost:56268 for channel out
2022-06-23 11:49:55,165 INFO  [io.sma.rea.mes.rabbitmq] (Quarkus Main Thread) SRMSG17036: RabbitMQ broker configured to localhost:56268 for channel coroutine
2022-06-23 11:49:55,167 INFO  [io.sma.rea.mes.rabbitmq] (Quarkus Main Thread) SRMSG17006: Establishing connection with RabbitMQ broker for channel `coroutine`
2022-06-23 11:49:55,170 INFO  [io.sma.rea.mes.rabbitmq] (Quarkus Main Thread) SRMSG17007: Connection with RabbitMQ broker established for channel `coroutine`
2022-06-23 11:49:55,170 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (Quarkus Main Thread) Starting rabbitmq client
2022-06-23 11:49:55,176 INFO  [io.sma.rea.mes.rabbitmq] (Quarkus Main Thread) SRMSG17036: RabbitMQ broker configured to localhost:56268 for channel reactive
2022-06-23 11:49:55,177 INFO  [io.sma.rea.mes.rabbitmq] (Quarkus Main Thread) SRMSG17006: Establishing connection with RabbitMQ broker for channel `reactive`
2022-06-23 11:49:55,178 INFO  [io.sma.rea.mes.rabbitmq] (Quarkus Main Thread) SRMSG17007: Connection with RabbitMQ broker established for channel `reactive`
2022-06-23 11:49:55,178 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (Quarkus Main Thread) Starting rabbitmq client
2022-06-23 11:49:55,181 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (Quarkus Main Thread) Starting rabbitmq client
2022-06-23 11:49:55,248 WARN  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-worker-thread-0) Failed to consume: : java.io.IOException
	at com.rabbitmq.client.impl.AMQChannel.wrap(AMQChannel.java:129)
	at com.rabbitmq.client.impl.AMQChannel.wrap(AMQChannel.java:125)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1384)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1297)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1239)
	at io.vertx.rabbitmq.impl.RabbitMQClientImpl.lambda$basicConsumer$12(RabbitMQClientImpl.java:300)
	at io.vertx.rabbitmq.impl.RabbitMQClientImpl.lambda$forChannel$41(RabbitMQClientImpl.java:810)
	at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=404, reply-text=NOT_FOUND - no queue 'coroutine' in vhost '/', class-id=60, method-id=20)
	at com.rabbitmq.utility.ValueOrException.getValue(ValueOrException.java:66)
	at com.rabbitmq.utility.BlockingValueOrException.uninterruptibleGetValue(BlockingValueOrException.java:36)
	at com.rabbitmq.client.impl.AMQChannel$BlockingRpcContinuation.getReply(AMQChannel.java:502)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1378)
	... 15 more
Caused by: com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=404, reply-text=NOT_FOUND - no queue 'coroutine' in vhost '/', class-id=60, method-id=20)
	at com.rabbitmq.client.impl.ChannelN.asyncShutdown(ChannelN.java:517)
	at com.rabbitmq.client.impl.ChannelN.processAsync(ChannelN.java:341)
	at com.rabbitmq.client.impl.AMQChannel.handleCompleteInboundCommand(AMQChannel.java:182)
	at com.rabbitmq.client.impl.AMQChannel.handleFrame(AMQChannel.java:114)
	at com.rabbitmq.client.impl.AMQConnection.readFrame(AMQConnection.java:739)
	at com.rabbitmq.client.impl.AMQConnection.access$300(AMQConnection.java:47)
	at com.rabbitmq.client.impl.AMQConnection$MainLoop.run(AMQConnection.java:666)
	... 1 more

2022-06-23 11:49:55,251 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-worker-thread-0) Stopping rabbitmq client
2022-06-23 11:49:55,253 INFO  [io.sma.rea.mes.rabbitmq] (vert.x-eventloop-thread-0) SRMSG17000: RabbitMQ Receiver listening address coroutine
2022-06-23 11:49:55,254 WARN  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-worker-thread-0) Failed to consume: : java.io.IOException
	at com.rabbitmq.client.impl.AMQChannel.wrap(AMQChannel.java:129)
	at com.rabbitmq.client.impl.AMQChannel.wrap(AMQChannel.java:125)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1384)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1297)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1239)
	at io.vertx.rabbitmq.impl.RabbitMQClientImpl.lambda$basicConsumer$12(RabbitMQClientImpl.java:300)
	at io.vertx.rabbitmq.impl.RabbitMQClientImpl.lambda$forChannel$41(RabbitMQClientImpl.java:810)
	at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=404, reply-text=NOT_FOUND - no queue 'reactive' in vhost '/', class-id=60, method-id=20)
	at com.rabbitmq.utility.ValueOrException.getValue(ValueOrException.java:66)
	at com.rabbitmq.utility.BlockingValueOrException.uninterruptibleGetValue(BlockingValueOrException.java:36)
	at com.rabbitmq.client.impl.AMQChannel$BlockingRpcContinuation.getReply(AMQChannel.java:502)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1378)
	... 15 more
Caused by: com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=404, reply-text=NOT_FOUND - no queue 'reactive' in vhost '/', class-id=60, method-id=20)
	at com.rabbitmq.client.impl.ChannelN.asyncShutdown(ChannelN.java:517)
	at com.rabbitmq.client.impl.ChannelN.processAsync(ChannelN.java:341)
	at com.rabbitmq.client.impl.AMQChannel.handleCompleteInboundCommand(AMQChannel.java:182)
	at com.rabbitmq.client.impl.AMQChannel.handleFrame(AMQChannel.java:114)
	at com.rabbitmq.client.impl.AMQConnection.readFrame(AMQConnection.java:739)
	at com.rabbitmq.client.impl.AMQConnection.access$300(AMQConnection.java:47)
	at com.rabbitmq.client.impl.AMQConnection$MainLoop.run(AMQConnection.java:666)
	... 1 more

2022-06-23 11:49:55,255 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-worker-thread-0) Stopping rabbitmq client
2022-06-23 11:49:55,255 INFO  [io.sma.rea.mes.rabbitmq] (vert.x-eventloop-thread-0) SRMSG17000: RabbitMQ Receiver listening address reactive
2022-06-23 11:49:55,263 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-eventloop-thread-0) Starting rabbitmq client
2022-06-23 11:49:55,269 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-eventloop-thread-0) Starting rabbitmq client
2022-06-23 11:49:55,276 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-eventloop-thread-0) Successed to restart client. 
2022-06-23 11:49:55,281 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-eventloop-thread-0) Successed to restart client. 
2022-06-23 11:49:55,285 ERROR [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-eventloop-thread-0) Failed to restart consumer: : java.io.IOException
	at com.rabbitmq.client.impl.AMQChannel.wrap(AMQChannel.java:129)
	at com.rabbitmq.client.impl.AMQChannel.wrap(AMQChannel.java:125)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1384)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1297)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1239)
	at io.vertx.rabbitmq.impl.RabbitMQClientImpl.lambda$null$2(RabbitMQClientImpl.java:200)
	at io.vertx.rabbitmq.impl.RabbitMQClientImpl.lambda$forChannel$41(RabbitMQClientImpl.java:810)
	at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=404, reply-text=NOT_FOUND - no queue 'coroutine' in vhost '/', class-id=60, method-id=20)
	at com.rabbitmq.utility.ValueOrException.getValue(ValueOrException.java:66)
	at com.rabbitmq.utility.BlockingValueOrException.uninterruptibleGetValue(BlockingValueOrException.java:36)
	at com.rabbitmq.client.impl.AMQChannel$BlockingRpcContinuation.getReply(AMQChannel.java:502)
	at com.rabbitmq.client.impl.ChannelN.basicConsume(ChannelN.java:1378)
	... 15 more
Caused by: com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=404, reply-text=NOT_FOUND - no queue 'coroutine' in vhost '/', class-id=60, method-id=20)
	at com.rabbitmq.client.impl.ChannelN.asyncShutdown(ChannelN.java:517)
	at com.rabbitmq.client.impl.ChannelN.processAsync(ChannelN.java:341)
	at com.rabbitmq.client.impl.AMQChannel.handleCompleteInboundCommand(AMQChannel.java:182)
	at com.rabbitmq.client.impl.AMQChannel.handleFrame(AMQChannel.java:114)
	at com.rabbitmq.client.impl.AMQConnection.readFrame(AMQConnection.java:739)
	at com.rabbitmq.client.impl.AMQConnection.access$300(AMQConnection.java:47)
	at com.rabbitmq.client.impl.AMQConnection$MainLoop.run(AMQConnection.java:666)
	... 1 more

2022-06-23 11:49:55,287 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-worker-thread-0) Reconsume queue: reactive success
2022-06-23 11:49:55,294 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (Quarkus Main Thread) Stopping rabbitmq client
2022-06-23 11:49:55,564 INFO  [org.ecl.jet.uti.log] (Quarkus Main Thread) Logging initialized @10319ms to org.eclipse.jetty.util.log.Slf4jLog
2022-06-23 11:49:55,731 INFO  [org.ecl.jet.ser.Server] (Quarkus Main Thread) jetty-9.4.46.v20220331; built: 2022-03-31T16:38:08.030Z; git: bc17a0369a11ecf40bb92c839b9ef0a8ac50ea18; jvm 17.0.2+8-LTS-86
2022-06-23 11:49:55,752 INFO  [org.ecl.jet.ser.han.ContextHandler] (Quarkus Main Thread) Started o.e.j.s.ServletContextHandler@7809efa8{/__admin,null,AVAILABLE}
2022-06-23 11:49:55,766 INFO  [org.ecl.jet.ser.han.Con.ROOT] (Quarkus Main Thread) RequestHandlerClass from context returned com.github.tomakehurst.wiremock.http.StubRequestHandler. Normalized mapped under returned 'null'
2022-06-23 11:49:55,766 INFO  [org.ecl.jet.ser.han.ContextHandler] (Quarkus Main Thread) Started o.e.j.s.ServletContextHandler@33af1e7b{/,null,AVAILABLE}
2022-06-23 11:49:55,774 INFO  [org.ecl.jet.ser.AbstractConnector] (Quarkus Main Thread) Started NetworkTrafficServerConnector@8748549{HTTP/1.1, (http/1.1, h2c)}{0.0.0.0:9000}
2022-06-23 11:49:55,775 INFO  [org.ecl.jet.ser.Server] (Quarkus Main Thread) Started @10530ms
2022-06-23 11:49:55,838 INFO  [io.quarkus] (Quarkus Main Thread) quarkus-rmq-playground 1.0.0-SNAPSHOT on JVM (powered by Quarkus 2.10.0.CR1) started in 9.690s. Listening on: http://localhost:8080
2022-06-23 11:49:55,838 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2022-06-23 11:49:55,839 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, kotlin, rest-client-reactive, rest-client-reactive-jackson, resteasy-reactive, resteasy-reactive-jackson, smallrye-context-propagation, smallrye-reactive-messaging, smallrye-reactive-messaging-rabbitmq, vertx]
2022-06-23 11:50:05,290 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-eventloop-thread-0) Stopping rabbitmq client
2022-06-23 11:50:05,305 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-eventloop-thread-0) Starting rabbitmq client
2022-06-23 11:50:05,337 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-eventloop-thread-0) Successed to restart client. 
2022-06-23 11:50:05,343 INFO  [io.ver.rab.imp.RabbitMQClientImpl] (vert.x-worker-thread-0) Reconsume queue: coroutine success
```

</details>


After sending some messages you can see it handles some and then throws a `BackPressureFailure`

<details>

```log
2022-06-23 11:50:13,425 INFO  [io.sma.rea.mes.rabbitmq] (vert.x-eventloop-thread-0) SRMSG17033: A message sent to channel `reactive` has been ack'd
Sent 1 messages in 270 milliseconds
Sent 1 messages in 22 milliseconds
2022-06-23 11:50:16,335 INFO  [io.sma.rea.mes.rabbitmq] (vert.x-eventloop-thread-0) SRMSG17033: A message sent to channel `reactive` has been ack'd
2022-06-23 11:50:20,565 ERROR [io.sma.rea.mes.provider] (vert.x-eventloop-thread-0) SRMSG00201: Error caught while processing a message: io.smallrye.mutiny.subscription.BackPressureFailure: Buffer full, cannot emit item
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapMainSubscriber.failOverflow(MultiFlatMapOp.java:541)
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapMainSubscriber.tryEmit(MultiFlatMapOp.java:243)
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapInner.onItem(MultiFlatMapOp.java:607)
	at io.smallrye.mutiny.operators.multi.MultiMapOp$MapProcessor.onItem(MultiMapOp.java:50)
	at io.smallrye.mutiny.operators.multi.MultiMapOp$MapProcessor.onItem(MultiMapOp.java:50)
	at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
	at io.smallrye.mutiny.vertx.MultiReadStream.lambda$subscribe$2(MultiReadStream.java:77)
	at io.vertx.rabbitmq.impl.RabbitMQConsumerImpl.lambda$handler$0(RabbitMQConsumerImpl.java:59)
	at io.vertx.core.streams.impl.InboundBuffer.handleEvent(InboundBuffer.java:240)
	at io.vertx.core.streams.impl.InboundBuffer.write(InboundBuffer.java:130)
	at io.vertx.rabbitmq.impl.RabbitMQConsumerImpl.handleMessage(RabbitMQConsumerImpl.java:150)
	at io.vertx.rabbitmq.impl.QueueConsumerHandler.lambda$handleDelivery$0(QueueConsumerHandler.java:39)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:63)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:503)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)

2022-06-23 11:50:20,568 ERROR [io.sma.rea.mes.provider] (vert.x-eventloop-thread-0) SRMSG00201: Error caught while processing a message: io.smallrye.mutiny.subscription.BackPressureFailure: Buffer full, cannot emit item
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapMainSubscriber.failOverflow(MultiFlatMapOp.java:541)
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapMainSubscriber.tryEmit(MultiFlatMapOp.java:243)
	at io.smallrye.mutiny.operators.multi.MultiFlatMapOp$FlatMapInner.onItem(MultiFlatMapOp.java:607)
	at io.smallrye.mutiny.operators.multi.MultiMapOp$MapProcessor.onItem(MultiMapOp.java:50)
	at io.smallrye.mutiny.operators.multi.MultiMapOp$MapProcessor.onItem(MultiMapOp.java:50)
	at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
	at io.smallrye.mutiny.vertx.MultiReadStream.lambda$subscribe$2(MultiReadStream.java:77)
	at io.vertx.rabbitmq.impl.RabbitMQConsumerImpl.lambda$handler$0(RabbitMQConsumerImpl.java:59)
	at io.vertx.core.streams.impl.InboundBuffer.handleEvent(InboundBuffer.java:240)
	at io.vertx.core.streams.impl.InboundBuffer.write(InboundBuffer.java:130)
	at io.vertx.rabbitmq.impl.RabbitMQConsumerImpl.handleMessage(RabbitMQConsumerImpl.java:150)
	at io.vertx.rabbitmq.impl.QueueConsumerHandler.lambda$handleDelivery$0(QueueConsumerHandler.java:39)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:63)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:38)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:503)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

</details

--- a/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
@@ -463,24 +463,27 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
 
         return establishExchange(client, ic)
                 .onItem().transform(v -> Boolean.TRUE.equals(ic.getQueueDeclare()) ? null : queueName)
-                .onItem().ifNull().switchTo(
-                        () -> {
-                            String serverQueueName = serverQueueName(queueName);
+                // Not declaring the queue, so validate its existence...
+                // Ensures RabbitMQClient is notified of invalid queues during connection cycle.
+                .onItem().ifNotNull().call(name -> client.messageCount(name).onFailure().invoke(log::unableToConnectToBroker))
+                // Declare the queue.
+                .onItem().ifNull().switchTo(() -> {
+                    String serverQueueName = serverQueueName(queueName);
 
-                            Uni<AMQP.Queue.DeclareOk> declare;
-                            if (serverQueueName.isEmpty()) {
-                                declare = client.queueDeclare(serverQueueName, false, true, true);
-                            } else {
-                                declare = client.queueDeclare(serverQueueName, ic.getQueueDurable(),
-                                        ic.getQueueExclusive(), ic.getQueueAutoDelete(), queueArgs);
-                            }
+                    Uni<AMQP.Queue.DeclareOk> declare;
+                    if (serverQueueName.isEmpty()) {
+                        declare = client.queueDeclare(serverQueueName, false, true, true);
+                    } else {
+                        declare = client.queueDeclare(serverQueueName, ic.getQueueDurable(),
+                                ic.getQueueExclusive(), ic.getQueueAutoDelete(), queueArgs);
+                    }
 
-                            return declare
-                                    .onItem().invoke(() -> log.queueEstablished(queueName))
-                                    .onFailure().invoke(ex -> log.unableToEstablishQueue(queueName, ex))
-                                    .onItem().transformToMulti(v -> establishBindings(client, ic))
-                                    .onItem().ignoreAsUni().onItem().transform(v -> queueName);
-                        });
+                    return declare
+                            .onItem().invoke(() -> log.queueEstablished(queueName))
+                            .onFailure().invoke(ex -> log.unableToEstablishQueue(queueName, ex))
+                            .onItem().transformToMulti(v -> establishBindings(client, ic))
+                            .onItem().ignoreAsUni().onItem().transform(v -> queueName);
+                });
     }
 
     /**

--- a/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
+++ b/smallrye-reactive-messaging-rabbitmq/src/main/java/io/smallrye/reactive/messaging/rabbitmq/RabbitMQConnector.java
@@ -37,7 +37,6 @@ import com.rabbitmq.client.impl.CredentialsProvider;
 
 import io.smallrye.mutiny.Multi;
 import io.smallrye.mutiny.Uni;
-import io.smallrye.mutiny.operators.multi.processors.BroadcastProcessor;
 import io.smallrye.mutiny.tuples.Tuple2;
 import io.smallrye.reactive.messaging.annotations.ConnectorAttribute;
 import io.smallrye.reactive.messaging.health.HealthReport;
@@ -189,21 +188,9 @@ public class RabbitMQConnector implements IncomingConnectorFactory, OutgoingConn
                 .map(String::trim).collect(Collectors.toList());
         log.receiverListeningAddress(queueName);
 
-        // The processor is used to inject AMQP Connection failure in the stream and trigger a retry.
-        BroadcastProcessor<Message<?>> processor = BroadcastProcessor.create();
-        receiver.exceptionHandler(t -> {
-            log.receiverError(t);
-            processor.onError(t);
-        });
-
-        return Multi.createFrom().deferred(
-                () -> {
-                    Multi<Message<?>> stream = receiver.toMulti()
-                            .map(m -> new IncomingRabbitMQMessage<>(m, holder, isTracingEnabled, onNack, onAck,
-                                    contentTypeOverride))
-                            .map(m -> isTracingEnabled ? TracingUtils.addIncomingTrace(m, queueName, attributeHeaders) : m);
-                    return Multi.createBy().merging().streams(stream, processor);
-                });
+        return receiver.toMulti()
+                .map(m -> new IncomingRabbitMQMessage<>(m, holder, isTracingEnabled, onNack, onAck, contentTypeOverride))
+                .map(m -> isTracingEnabled ? TracingUtils.addIncomingTrace(m, queueName, attributeHeaders) : m);
     }
 
     /**

